_includes/specials/taproot/en/01-single-sig.md

```diff
@@ -0,0 +1,75 @@
+Using Optech's [transaction size calculator][], we can compare the sizes
+of different types of single-sig transactions.  As expected,
+transactions using P2WPKH inputs and outputs are much smaller than those
+using P2PKH inputs and outputs---but, perhaps surprisingly, P2TR
+transactions are slightly larger than equivalent P2WPKH transactions.
+
+|                    | P2PKH (legacy) | P2WPKH (segwit v0) | P2TR (taproot/segwit v1) |
+|--------------------|----------------|--------------------|--------------------------|
+| **Output**         |  34            | 31                 | 43                       |
+| **Input**          |  148           | 67.75              | 57.25                    |
+| **2-in, 2-out tx** |  374           | 208.25             | 211.25                   |
```

Comment:
Wow, thanks for the amazing fact checking!  Looking at the source code for https://bitcoinops.org/en/tools/calc-size/ and comparing it to your B.SE answer, it appears I omitted the `count witness items` from all the segwit types, which explains both discrepancies above.  Will fix!

Reply:
Oh, good. I think I had looked at that before and hadn't noticed. :flushed:
It's one of those seldom but gratifying instances where intimate familiarity with input sizes is actually useful. :laughing:

Reply:
Oh, when I went to edit the code, I discovered it does seem to be correct.  We count `count witness items` as part of the transaction overhead (since it's only used once per tx, not per input).  So I think both the P2WPKH and P2TR examples are correct as written.

Reply:
No, `count witness item` is definitely per input, not per transaction. For P2WPKH it is two (signature and public key), for P2TR it is one (signature only).

Interesting thing I learned some time ago, later than I should have: there must be a witness stack for every input as soon there is a single segwit input. For non-segwit inputs this is just a 0 for the `count witness items`. This actually causes an input size non-linearity that had baffled me for a while then, since adding the first segwit input increases the whole transaction weight by more than just the segwit input's weight.

Reply:
> No, count witness item is definitely per input, not per transaction.

Huh, today I learned.  It looks like I only skimmed BIP141, which gives a misleading impression with:

> `[nVersion][marker][flag][txins][txouts][witness][nLockTime]`

(How come `txins` and `txouts` are plural, but `witness` is singular?)

But reading a bit further, it does say:

> Each txin is associated with a witness field.

I have some asserts to actual transactions in the code for the tool, but it looks like I only selected 1-input examples and so they didn't catch this.  I'll find a 2-input example, add it, and update the code.  Thanks so much for (1) discovering the mistake, (2) putting up with my vacillating, (3) continuing to try to show me the truth until I saw the light!

Reply:
Should be fixed now (two separate commits; one for the tx tool; one for the newsletter text).  Thanks again!

Reply:
All good, glad to return the favor after learning so much from your writings. :+1: